Source/EasyNetQ/MessageVersioning/MessageVersionStack.cs

```diff
@@ -56,11 +56,18 @@ private static Stack<Type> ExtractMessageVersions( Type type )
 
         private static Type GetSupersededType( Type type )
         {
-             return type
+            var types = FindSupersedes(type);
+            var parentTypes = FindSupersedes(type.BaseType);
+
+            return types.FirstOrDefault(t => !parentTypes.Contains(t));
```

Comment:
I think `types.Except(parentTypes).FirstOrDefault()` will be much better here.

Reply:
Good point! 👍